File: Java/src/main/java/com/interviewkata/Santa.java
package com.interviewkata;

import java.util.Arrays;
import java.util.List;

public class Santa {
    public List<Gift> gifts = Arrays.asList(new Gift("Trencito"), new Gift("Bebote"));

    public void deliverGift(Child child) {
        Gift gift = null;
        if (child.getDesiredGift() == null) {
            gift = new Gift("Chocolate");
        } else {
            for (Gift giftInTheBag : gifts) {
                if (giftInTheBag == child.getDesiredGift()) {
                    gift = giftInTheBag;
                    break;
                }
            }
            if (gift == null) {
                gift = new Gift("Chocolate");
            }
            gifts.remove(gift);
        }

        child.setObtainedGift(gift);
    }

    public List<Gift> getGifts() {
        return gifts;
    }

    public void setGifts(List<Gift> gifts) {
        this.gifts = gifts;
    }
}

File: Kotlin/src/main/java/com/interviewkata/Santa.kt
package com.interviewkata

class Santa {
    var gifts =  mutableListOf(Gift("Trencito"), Gift("Bebote"))

    fun deliverGift(child: Child) {
        var gift: Gift? = null
        if (child.desiredGift == null) {
            gift = Gift("Chocolate")
        } else {
            gifts.forEach { giftInTheBag ->
                if (giftInTheBag == child.desiredGift) {
                    gift = giftInTheBag
                }
            }
            if (gift == null) {
                gift = Gift("Chocolate")
            }
            gifts.remove(gift!!)
        }
        child.obtainedGift = gift
    }
}
